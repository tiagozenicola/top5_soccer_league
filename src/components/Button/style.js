import styled from 'styled-components'


const StyledButton = styled.button`
  color: blue;
  border: none;
  background: none;
  text-decoration: underline;
  cursor: pointer;
`


export { StyledButton };
