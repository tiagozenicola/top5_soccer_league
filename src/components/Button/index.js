import React from 'react';
import { StyledButton } from './style';


const Button = (props) => {
  const {sortProperty, value, name, onClick, orientation} = props

  return <StyledButton onClick={() => onClick(name)} >
      {value}
      {
        sortProperty !== name ? '' : (  
          orientation === 1 ? '⇧' : '⇩'
        )
      }
    </StyledButton>
}


export default Button;
