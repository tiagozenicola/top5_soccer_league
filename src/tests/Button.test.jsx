import React from 'react'
import renderer from 'react-test-renderer'
import enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import Button from '../components/Button'
import 'jest-styled-components'

enzyme.configure({ adapter: new Adapter() })

it('renders Button with no sort', () => {
  const button = renderer.create(<Button value="P" name="position" sortProperty="percent" orientation={-1} onClick={() => {}}/>).toJSON()
  expect(button).toMatchSnapshot()
})

it('renders Button sorting asc', () => {
  const button = renderer.create(<Button value="P" name="position" sortProperty="position" orientation={-1} onClick={() => {}}/>).toJSON()
  expect(button).toMatchSnapshot()
})

it('renders Button sorting desc', () => {
  const button = renderer.create(<Button value="P" name="position" sortProperty="position" orientation={1} onClick={() => {}}/>).toJSON()
  expect(button).toMatchSnapshot()
})

it('test that click calls my function', () => {
  const sort = jest.fn();
  const button = shallow(<Button value="P" name="position" sortProperty="position" orientation={1} onClick={sort}/>)
  button.simulate("click");
  expect(sort).toHaveBeenCalledTimes(1);
})