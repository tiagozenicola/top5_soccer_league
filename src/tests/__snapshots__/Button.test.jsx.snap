// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button sorting asc 1`] = `
.c0 {
  color: blue;
  border: none;
  background: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

<button
  className="c0"
  onClick={[Function]}
>
  P
  ⇩
</button>
`;

exports[`renders Button sorting desc 1`] = `
.c0 {
  color: blue;
  border: none;
  background: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

<button
  className="c0"
  onClick={[Function]}
>
  P
  ⇧
</button>
`;

exports[`renders Button with no sort 1`] = `
.c0 {
  color: blue;
  border: none;
  background: none;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  cursor: pointer;
}

<button
  className="c0"
  onClick={[Function]}
>
  P
  
</button>
`;
